subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java

```diff
@@ -315,8 +316,11 @@ private void configureArchivesAndComponent(Project project, JavaPluginConvention
         addJar(runtimeConfiguration, jarArtifact);
         addRuntimeVariants(runtimeElementsConfiguration, jarArtifact, javaCompile, processResources);
 
-        project.getComponents().add(objectFactory.newInstance(JavaLibrary.class, project.getConfigurations(), jarArtifact));
-        project.getComponents().add(objectFactory.newInstance(JavaLibraryPlatform.class, project.getConfigurations()));
+        // Currently, we publish capabilities as they are declared unconditionally. In the future,
+        // We might want to default to those, but allow overriding
```

Comment:
I don't think this comment is required: this is no different from anything declared in `dependencies`: we publish this stuff directly but ideally the component could be modified. Any change to published metadata should also apply to the dependencies/constraints/capabilities that are obtained via a project dependency.